### PR TITLE
Configurable Decompression Methods on Http Request

### PR DESCRIPTION
**Pull request by jchannon**
This is a PR from a commit that @fabman08 did on issue #561 on his own fork but never submitted a PR.

We are having an issue with a bug in Mono and need to turn off gzip support ASAP 😄  👍 

Thanks

### CODE REVIEW

File: RestSharp.Net45/RestSharp.Net45.csproj [hunk 1]
```diff
@@ -277,7 +277,7 @@
     </None>
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <Import Project="$(SolutionDir)\.nuget\nuget.targets" />
+  <!-- <Import Project="$(SolutionDir)\.nuget\nuget.targets" /> -->
```
Comment (hallem) [inline RestSharp.Net45/RestSharp.Net45.csproj]: why was this changed?

File: RestSharp/Http.Sync.cs [hunk 2]
```diff
@@ -25,6 +25,8 @@
 
 namespace RestSharp
 {
+    using RestSharp.Authenticators.OAuth.Extensions;
```
Comment (hallem) [inline RestSharp/Http.Sync.cs]: using statements go outside the namespace

**Comment by hallem** [comment]
this looks like it would be a breaking change.  you'd have to set the current value to be the default unless it's explicitly set.  you'll also need to add some tests to make sure this is working.

**Comment by Fabman08** [comment]
@hallem I'm sorry, this was my fist PR :(
I'll fix it as soon as I can also adding unit tests

**Comment by hallem** [comment]
no problem we all start somewhere.  i would have done the same thing, it takes some time to notice breaking changes and be aware of them.  the rest of the code looks good though.

**Comment by jchannon** [comment]
Brill!

Not sure if you want to do your own PR or fork my branch and do a PR to
that which will then auto update this PR

**Comment by Fabman08** [comment]
@jchannon I think I'll fork your branch so it will be auto updated :)

**Comment by jchannon** [comment]
I've made it easier and made you a contributor on my fork, so clone and
change branch and push directly, no need for PRs and it will then update
this PR.

**Comment by jchannon** [comment]
@Fabman08 Any idea when you're going to get time to look at this?

**Comment by Fabman08** [comment]
@jchannon I'm doing it right now :)
I've a question: where can I put the unit test? RestSharp.Tests or RestSharp.Test.Signed ?
Thank you

**Comment by jchannon** [comment]
Unless you're adding new files then RestSharp.Tests.csproj and
Retsharp.Tests.Signed.csproj will pick the changes up automatically

**Comment by dalpert-korewireless** [comment]
this would be super useful

**Comment by Fabman08** [comment]
@jchannon I've added two simple unit test, finally! :) Is it enough or you need more complex and specific tests?
Thank you!

**Comment by jchannon** [comment]
we've moved on i'm afraid and away from mono and restsharp

**Comment by alexeyzimarev** [comment]
@Fabman08 would be nice if you remove comments from the code you submitted.

**Comment by Fabman08** [comment]
@alexeyzimarev done! :)

**Comment by wangengzheng** [comment]
client.ConfigureWebRequest(wr=>{
      wr.AutomaticDecompression = System.Net.DecompressionMethods.None;
});
